Formula/sratoolkit.rb

```diff
@@ -37,93 +28,54 @@ class Sratoolkit < Formula
-  depends_on "hdf5@1.10"
-  depends_on "libmagic"
-
-  uses_from_macos "perl" => :build
+  depends_on "hdf5"
+
   uses_from_macos "libxml2"
 
+  # Modify cmake scripts to avoid building tests when BUILD_TESTING=OFF and
+  # to install into CMAKE_INSTALL_LIBDIR rather than CMAKE_INSTALL_PREFIX/lib64.
+  # Issue ref: https://github.com/ncbi/sra-tools/issues/638
+  # Issue ref: https://github.com/ncbi/sra-tools/issues/639
+  patch :DATA
+
   def install
-    libxml2_prefix = if OS.mac?
-      MacOS.sdk_path/"usr"
-    else
-      Formula["libxml2"].opt_prefix
-    end
-    with_formula_args = %W[
-      --with-hdf5-prefix=#{Formula["hdf5@1.10"].opt_prefix}
-      --with-magic-prefix=#{Formula["libmagic"].opt_prefix}
-      --with-xml2-prefix=#{libxml2_prefix}
-    ]
-
-    ngs_sdk_prefix = buildpath/"ngs-sdk-prefix"
-    resource("ngs-sdk").stage do
-      cd "ngs-sdk" do
-        system "./configure", "--prefix=#{ngs_sdk_prefix}",
-                              "--build=#{buildpath}/ngs-sdk-build"
-        system "make"
-        system "make", "install"
-      end
-    end
+    (buildpath/"ncbi-vdb-source").install resource("ncbi-vdb")
 
-    ncbi_vdb_source = buildpath/"ncbi-vdb-source"
-    ncbi_vdb_build = buildpath/"ncbi-vdb-build"
-    ncbi_vdb_source.install resource("ncbi-vdb")
-    cd ncbi_vdb_source do
-      # Fix detection of hdf5 library on macOS as Apple Clang linker doesn't
-      # allow mixing static (-Wl,-Bstatic) and dynamic (-Wl,-Bdynamic) libraries
-      inreplace "setup/konfigure.perl", "-Wl,-Bstatic -lhdf5 -Wl,-Bdynamic", "-lhdf5" if OS.mac?
-
-      # Fix Linux error: `pshufb' is not supported on `generic64.aes'
-      # Upstream ref: https://github.com/ncbi/ncbi-vdb/issues/14
-      inreplace "libs/krypto/Makefile", "-Wa,-march=generic64+aes", "" if OS.linux?
-
-      system "./configure", "--prefix=#{buildpath}/ncbi-vdb-prefix",
-                            "--build=#{ncbi_vdb_build}",
-                            "--with-ngs-sdk-prefix=#{ngs_sdk_prefix}",
-                            *with_formula_args
-      ENV.deparallelize { system "make" }
-    end
+    # Workaround to allow clang/aarch64 build to use the gcc/arm64 directory
+    # Issue ref: https://github.com/ncbi/ncbi-vdb/issues/65
+    ln_s "../gcc/arm64", buildpath/"ncbi-vdb-source/interfaces/cc/clang/aarch64" if Hardware::CPU.arm?
 
-    # Fix the error: ld: library not found for -lmagic-static
-    # Upstream PR: https://github.com/ncbi/sra-tools/pull/105
-    inreplace "tools/copycat/Makefile", "-smagic-static", "-smagic"
+    # Workaround to remove hardcoded bitmagic SSE4.2 optimization if needed
+    if !Hardware::CPU.intel? || !Hardware::CPU.sse4_2? || (build.bottle? && !MacOS.version.requires_sse42?)
+      bitmagic_opt = Hardware::CPU.arm? ? "-DDBMNEONOPT" : "-DBMSSE2OPT"
+      inreplace "tools/sharq/CMakeLists.txt", "add_definitions(-msse4.2 -DBMSSE42OPT)",
+                                              "add_definitions(#{bitmagic_opt})"
```

Comment:
Not sure about `Hardware::CPU.sse4_2?` usage since didn't see any formulae using it. Mainly want to allow source-builds to keep SSE4.2 if available. There is also AVX2 optimization option, but ignoring for now.

Side note is that Linux currently doesn't build `sharq` as it needs `zlib` and `bzip2`. However, adding these means we need to cannot use `gcc-5` as build requires C++17. Didn't seem worth updating to forcing `gcc` dependency right now.